packages/main/src/files/file-manager.ts

```diff
@@ -42,23 +42,36 @@ export interface UploadMetadata {
  * @public
  */
 export class GoogleAIFileManager {
+  private _requestOptions: RequestOptions;
   constructor(
     public apiKey: string,
-    private _requestOptions?: RequestOptions,
-  ) {}
+    requestOptions?: RequestOptions,
```

Comment:
I don't think we should have a type that says they can pass an abortSignal property but then we immediately delete it, we should probably make a new interface, maybe GlobalRequestOptions or something. Or rename the requestOptions that can be passed to individual methods as SingleRequestOptions or something.

For non-TS users, I think it may helpful to have a warning here if they accidentally provide it. I don't think we should delete it silently. You'll have to wrestle TS a little to provide the warning if you separate into two types but I think that's reasonable as it's to accommodate non-TS users.

Reply:
I couldn't find a proper name to replace `RequestOptions` which encompasses `FileManager`, `ChatSession`,  and `GenerativeModel` which all use `RequestOptions` so I've made a `SingleRequestOptions` instead.